### PR TITLE
Add asset signatories to newly created asset deposits

### PR DESCRIPTION
**Pull request by Sarahb-da**
When an asset deposit is created, the signatories of that asset are added as observers. That way any issuer can view allocations of their asset.

### CODE REVIEW

File: daml/Marketplace/Rule/AllocationAccount.daml [hunk 1]
```diff
@@ -38,7 +39,14 @@ template AllocationAccountRule
...
       creditAccount : Account -> Asset -> Update (ContractId AssetDeposit)
-      creditAccount = \account asset -> exerciseByKey @AssetSettlementRule account.id AssetSettlement_Credit with ctrl = nominee, ..
+      creditAccount = \account asset -> do
+        depositCid <- exerciseByKey @AssetSettlementRule account.id AssetSettlement_Credit with ctrl = nominee, ..
+        deposit <- fetch depositCid
+        let
+          addProvider    = exerciseByKey @AssetSettlementRule account.id AssetSettlement_AddController with ctrl = deposit.account.owner
+          removeProvider = exerciseByKey @AssetSettlementRule account.id AssetSettlement_RemoveController with ctrl = deposit.account.owner
+          addObservers   = exercise depositCid AssetDeposit.AssetDeposit_SetObservers with newObservers = deposit.asset.id.signatories `union` deposit.observers
+        addProvider *> addObservers <* removeProvider
```
Comment (Sarahb-da) [inline daml/Marketplace/Rule/AllocationAccount.daml:49]: this causes an error in E2E test bc it lacks authorization from Alice

.. so still workin on this part

File: daml/Marketplace/Rule/AllocationAccount.daml [hunk 2]
```diff
@@ -40,14 +40,15 @@ template AllocationAccountRule
       creditAccount : Account -> Asset -> Update (ContractId AssetDeposit)
       creditAccount = \account asset -> exerciseByKey @AssetSettlementRule account.id AssetSettlement_Credit with ctrl = nominee, ..
```
Comment (alexgraham-da) [inline daml/Marketplace/Rule/AllocationAccount.daml:41]: Did you mean to take out the `AssetDeposit_SetObservers` code here?

File: daml/Marketplace/Rule/AllocationAccount.daml [hunk 3]
```diff
@@ -40,14 +40,15 @@ template AllocationAccountRule
...
 
-    controller nominee, account.owner can
-      nonconsuming Transfer : ContractId AssetDeposit
+    nonconsuming choice Transfer : ContractId AssetDeposit
         with
           transferTo : Account
           depositCid : ContractId AssetDeposit
+        controller nominee, account.owner, transferTo.owner
```
Comment (brian-weir) [inline daml/Marketplace/Rule/AllocationAccount.daml:47]: Have you tested that the trading and the auction settlements are still both working after adding `transferTo.owner` to this controller ?

**Review by brian-weir — APPROVED** [review]
lgtm! 👍

File: daml/Marketplace/Rule/AllocationAccount.daml (hunk 2 above)
Comment (Sarahb-da) [reply daml/Marketplace/Rule/AllocationAccount.daml]: Nope. Added back

File: daml/Marketplace/Rule/AllocationAccount.daml (hunk 3 above)
Comment (Sarahb-da) [reply daml/Marketplace/Rule/AllocationAccount.daml:50]: Yes

**Review by alexgraham-da — APPROVED** [review]
looks good to me!

**Pull request merged**